### ISSUE TITLE
[SD-NG] implement scale theme

### ISSUE THREAD

**Issue by Aloento** [stackmon/otc-status-dashboard#40]
Should improve the auto-resizing auto-proportion of text/pictures

Consult with Tino/Sebastian and evantually check feasibility for scale theme
It should appear in https://github.com/stackmon/otc-status-dashboard

**Comment by Aloento** [comment]
@vladimirhasko makes the issue title and first description. My task is to investigate the possibility.
So, I post my founds below.

---

This issue mentioned two different demands:

1. reproduce the Scale design.
2. flex layout.

## Flex layout

Auto-fitting the client's screen size can be done quickly using `display: flex` and related properties.
The proportion of the image can be set to `aspect-ratio`.

Moreover, **using `tailwind css` is better than writing ourselves.**

## Scale design

The scale design can be introduced by importing `scale-components.css`.
Let's give the `button` as an example.

### Chang one button's CSS

We can do something like

```css
.button {
  ...
  font-weight: var(--font-weight);
  font-size: var(--font-size);
  line-height: var(--line-height);
  min-height: var(--min-height);
  min-width: var(--min-width);
  ...
}
```

We also need to consider the variance, like

```css
.button:after {...}
.button:not(.button--disabled):focus {...}
.button--disabled {...}
...
```

### Make component to reuse

We can do simply like 

```html
 <button
    class="button"
    ...
 />
```

to apply our style to one button, or we can use `Type Selectors` to apply to all buttons.

But sometimes, the button may contain an icon or some sub-elements, and then every time, we need to copy all the code if we want to make the same thing, which is redundant.

So, we need to make it into the template or a component to reuse it.

### Then

We should repeat the above steps for each component we need to reuse.
In the end, we have successfully created another Scale in Flask.
**Which is reinventing the wheel.**

### Another way

Let's think about what will happen if we directly import Scale.

To achieve the aim of "everyone can use SD2 rather than OTC only", we should treat Scale as any ordinary open source library by using the particular version `scale-components-neutral`.

Then, just use it. In theory, everyone could write some code that collects the API-based data from the endpoints and displays it there; this is not very difficult.

In a nutshell, the best way to implement a scale theme is just to use the Scale itself.

---

> I've heard someone trying to make SD2 compatible with IE6? That's not going to happen.
> In fact, the current SD2 uses **ES5** `bootstrap.js`, which is supported by IE10. It's the default browser for Windows 8.
> And Scale can be easily transformed to ES5 (it's ES6 by default) by importing a special version of `scale-components.cjs.js`.
> Also, the Let's Encrypt we use requires at least TLS 1.2, which is supported by Windows 7. So in terms of connectivity, the oldest we can consider is IE9.

**So, we should focus on the evergreens.**
This is a common practice among others like GCP, AWS, Github...

@anton-sidelnikov @Nils-Magnus, please give your opinion.
Thank you.

**Comment by Nils-Magnus** [comment]
If tailwind CSS is a help to calculate your screen layout, feel free to use it, but please keep the potential technical debts in mind, if you go this way: It may increase the number of moving parts of your application, you might have more colleagues in your calls asking you about the why and hows, but apart from that there's no objection from my side to it.

Scale is a huge framework and thus comes with some overhead upfront. The main question is how to compensate for this initial offset. Coming up with a Scale version of Status dashboard serves two purposes:
a) it provides our users and customers a modern UI that they find useful and convenient to use,
b) it is a training center for ourselves to further grow and collect XP on this very matter.
Therefore I opt to striving to "doing it right" right from the beginning. If this is combined with the greenfield approach, that's fine with me (as long as greenfield is not misinterpreted as "ridiculously unstable experimental tech-fantasies).
